examples/experimental/README.md

```diff
@@ -0,0 +1,70 @@
+RFC: Toward TF Encrypted 1.0
+==================================
+This is a living design doc for TF Encrypted's first big refactor.
+
+T.O.C.
+1. Problems with current implementation
+2. Overview
+3. Details
+4. Protocol Requirements
+
+
+### Problems with current implementation
+- Too much logic is wrapped up in the Pond protocol, and the file is too large
+- It is not clear which methods a new protocol needs to override to conform to the public TFE API (this API is largely implicitly defined and not stated explicitly anywhere)
+- BackingTensors being wrapped up in factories are impossible to test
+- Excessive code duplication: each operation is implemented at least once for each BackingTensor and each protocol
+
+
+
+### Overview
+The guiding philosophy of the refactor will be to replace TFE's major features with native TensorFlow abstractions that are covered by the TF Stability Guarantees for the lifetime of TF 2.0.  The goals of doing so include:
+- Upgrading TF Encrypted to TensorFlow 2.0
+- Leverage as much of the existing TensorFlow ecosystem with minimal rewriting and/or maintenance
+- Continue to rely on the stability guarantee of the TensorFlow Public API
+- Improve our own code modularity, extensibility, and sustainability and reduce code duplication
+
+Concretely, we will replace our own type system of custom tensors with a light set of hooks over existing tensor types in TensorFlow.  The current Protocol class would be replaced by custom subclasses of `tf.distribute.Strategy` and `tf.distribute.StrategyExtended` that define the necessary protocol operations and handle any distributed computation that's required by them.
+
+By doing so, we hope to satisfy each of the above goals, as well as all potential future requirements detailed by @mortendahl (see below)
+
+### Details
+- `Protocol` class replaced by custom implementations of `tf.distribute.Strategy` and `tf.distribute.StrategyExtended`
+  - The Strategy abstraction functions as our entry-point into the rest of the higher-level TF APIs
+  - The current strategy scope will determine which protocol kernel is used behind the scenes when calling a tfe op, e.g. `tfe.matmul`, similar to how it works now with Protocols
+  - For n-party protocols, this API provides a convenient set of tools for orchestrating distributed computation in a way that can induce out-of-the-box support for both tf.keras and the Estimator API
+  - Note these need not operate in the multi-party setting: see [`tf.distribute.OneDeviceStrategy`](https://www.tensorflow.org/api_docs/python/tf/distribute/OneDeviceStrategy) for an example of a single device strategy
+- Replace `BackingTensor` (`BackingVariable`) with `tf.Tensor` (`tf.Variable`)
+  - Allows us to use `Strategy` instead of `Protocol` (since Strategies must run on native TF Tensors/Variables)
```

Comment:
Strategies may put further limitations on the tensors, eg requiring certain dtypes of the tensors (this was the case when trying to get TF Big to work with Keras: https://github.com/tf-encrypted/tf-encrypted/issues/601#issuecomment-510487100

Reply:
by design, the base strategies do not have these restrictions, since they're meant to be extended -- the higher-level ones (e.g. MirroredStrategy) do, but that shouldn't matter for us.